In [3]:
# Basic Libraries
import numpy as np
import pandas as pd

# Problem 1

In [6]:
df = pd.read_csv('../Resources/train.csv')

In [12]:
df.shape

(1460, 81)

There are **1460** observations (rows).

There are **81** variables (columns).

In [17]:
df.dtypes

Id                 int64
MSSubClass         int64
MSZoning          object
LotFrontage      float64
LotArea            int64
                  ...   
MoSold             int64
YrSold             int64
SaleType          object
SaleCondition     object
SalePrice          int64
Length: 81, dtype: object

In [10]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1460 entries, 0 to 1459
Data columns (total 81 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Id             1460 non-null   int64  
 1   MSSubClass     1460 non-null   int64  
 2   MSZoning       1460 non-null   object 
 3   LotFrontage    1201 non-null   float64
 4   LotArea        1460 non-null   int64  
 5   Street         1460 non-null   object 
 6   Alley          91 non-null     object 
 7   LotShape       1460 non-null   object 
 8   LandContour    1460 non-null   object 
 9   Utilities      1460 non-null   object 
 10  LotConfig      1460 non-null   object 
 11  LandSlope      1460 non-null   object 
 12  Neighborhood   1460 non-null   object 
 13  Condition1     1460 non-null   object 
 14  Condition2     1460 non-null   object 
 15  BldgType       1460 non-null   object 
 16  HouseStyle     1460 non-null   object 
 17  OverallQual    1460 non-null   int64  
 18  OverallC

There are
**3** variables that are float, 
**35** variables that are integers, 
and **43** variables that are objects

In [19]:
pd.set_option('display.float_format', lambda x: '%.2f' % x)
df.describe()

Id  MSSubClass  LotFrontage   LotArea  OverallQual  OverallCond  \
count 1460.00     1460.00      1201.00   1460.00      1460.00      1460.00   
mean   730.50       56.90        70.05  10516.83         6.10         5.58   
std    421.61       42.30        24.28   9981.26         1.38         1.11   
min      1.00       20.00        21.00   1300.00         1.00         1.00   
25%    365.75       20.00        59.00   7553.50         5.00         5.00   
50%    730.50       50.00        69.00   9478.50         6.00         5.00   
75%   1095.25       70.00        80.00  11601.50         7.00         6.00   
max   1460.00      190.00       313.00 215245.00        10.00         9.00   

       YearBuilt  YearRemodAdd  MasVnrArea  BsmtFinSF1  ...  WoodDeckSF  \
count    1460.00       1460.00     1452.00     1460.00  ...     1460.00   
mean     1971.27       1984.87      103.69      443.64  ...       94.24   
std        30.20         20.65      181.07      456.10  ...      125.34   
min      1872.00       1950.00        0.00        0.00  ...        0.00   
25%      1954.00       1967.00        0.00        0.00  ...        0.00   
50%      1973.00       1994.00        0.00      383.50  ...        0.00   
75%      2000.00       2004.00      166.00      712.25  ...      168.00   
max      2010.00       2010.00     1600.00     5644.00  ...      857.00   

       OpenPorchSF  EnclosedPorch  3SsnPorch  ScreenPorch  PoolArea  MiscVal  \
count      1460.00        1460.00    1460.00      1460.00   1460.00  1460.00   
mean         46.66          21.95       3.41        15.06      2.76    43.49   
std          66.26          61.12      29.32        55.76     40.18   496.12   
min           0.00           0.00       0.00         0.00      0.00     0.00   
25%           0.00           0.00       0.00         0.00      0.00     0.00   
50%          25.00           0.00       0.00         0.00      0.00     0.00   
75%          68.00           0.00       0.00         0.00      0.00     0.00   
max         547.00         552.00     508.00       480.00    738.00 15500.00   

       MoSold  YrSold  SalePrice  
count 1460.00 1460.00    1460.00  
mean     6.32 2007.82  180921.20  
std      2.70    1.33   79442.50  
min      1.00 2006.00   34900.00  
25%      5.00 2007.00  129975.00  
50%      6.00 2008.00  163000.00  
75%      8.00 2009.00  214000.00  
max     12.00 2010.00  755000.00  

[8 rows x 38 columns]

The `.describe()` method lets us view relevant statistics about each variable (column) in the data.

# Problem 2

In [30]:
html_data = pd.read_html("https://en.wikipedia.org/wiki/2016_Summer_Olympics_medal_table")

In [31]:
len(html_data)

7

In [32]:
html_data[2]

Rank                   NOC  Gold  Silver  Bronze  Total
0                  1         United States    46      37      38    121
1                  2         Great Britain    27      23      17     67
2                  3                 China    26      18      26     70
3                  4                Russia    19      17      20     56
4                  5               Germany    17      10      15     42
..               ...                   ...   ...     ...     ...    ...
82                78               Nigeria     0       0       1      1
83                78              Portugal     0       0       1      1
84                78   Trinidad and Tobago     0       0       1      1
85                78  United Arab Emirates     0       0       1      1
86  Totals (86 NOCs)      Totals (86 NOCs)   307     307     359    973

[87 rows x 6 columns]

In [34]:
df = pd.DataFrame(html_data[2])

Rank            NOC  Gold  Silver  Bronze  Total
0    1  United States    46      37      38    121
1    2  Great Britain    27      23      17     67
2    3          China    26      18      26     70
3    4         Russia    19      17      20     56
4    5        Germany    17      10      15     42

In [36]:
new_df = df.head(20)

# Bonus Problem

In [41]:
df = pd.read_csv('../Resources/adult.data')

In [45]:
df.shape

(32560, 15)

In [46]:
df.info() 

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32560 entries, 0 to 32559
Data columns (total 15 columns):
 #   Column          Non-Null Count  Dtype 
---  ------          --------------  ----- 
 0   39              32560 non-null  int64 
 1    State-gov      32560 non-null  object
 2    77516          32560 non-null  int64 
 3    Bachelors      32560 non-null  object
 4    13             32560 non-null  int64 
 5    Never-married  32560 non-null  object
 6    Adm-clerical   32560 non-null  object
 7    Not-in-family  32560 non-null  object
 8    White          32560 non-null  object
 9    Male           32560 non-null  object
 10   2174           32560 non-null  int64 
 11   0              32560 non-null  int64 
 12   40             32560 non-null  int64 
 13   United-States  32560 non-null  object
 14   <=50K          32560 non-null  object
dtypes: int64(6), object(9)
memory usage: 3.7+ MB


In [47]:
df.describe()

39      77516       13     2174        0       40
count 32560.00   32560.00 32560.00 32560.00 32560.00 32560.00
mean     38.58  189781.81    10.08  1077.62    87.31    40.44
std      13.64  105549.76     2.57  7385.40   402.97    12.35
min      17.00   12285.00     1.00     0.00     0.00     1.00
25%      28.00  117831.50     9.00     0.00     0.00    40.00
50%      37.00  178363.00    10.00     0.00     0.00    40.00
75%      48.00  237054.50    12.00     0.00     0.00    45.00
max      90.00 1484705.00    16.00 99999.00  4356.00    99.00

No header is present. The first row became the header. Header=None is added.

In [71]:
df = pd.read_csv('../Resources/adult.data', header = None)
df.head(1)

0           1      2           3   4               5              6   \
0  39   State-gov  77516   Bachelors  13   Never-married   Adm-clerical   

               7       8      9     10  11  12              13      14  
0   Not-in-family   White   Male  2174   0  40   United-States   <=50K

In [103]:
dictionary = {}

for year in range(2000,2017,4):
    df = pd.read_html(f'https://en.wikipedia.org/wiki/{year}_Summer_Olympics_medal_table', index_col = 0)[2]
    dictionary[year] = df
    dictionary[f'{year}_20'] = df.head(20)
    
    print(f"Summer Olympics {year} Table".center(50), df.head(20), '_'*50, end='\n\n', sep='\n')

            Summer Olympics 2000 Table            
             Nation  Gold  Silver  Bronze  Total
Rank                                            
1     United States    37      24      32     93
2            Russia    32      28      29     89
3             China    28      16      14     58
4        Australia*    16      25      17     58
5           Germany    13      17      26     56
6            France    13      14      11     38
7             Italy    13       8      13     34
8       Netherlands    12       9       4     25
9              Cuba    11      11       7     29
10    Great Britain    11      10       7     28
11          Romania    11       6       9     26
12      South Korea     8      10      10     28
13          Hungary     8       6       3     17
14           Poland     6       5       3     14
15            Japan     5       8       5     18
16         Bulgaria     5       6       2     13
17           Greece     4       6       3     13
18           Swede